sql/src/main/antlr/OpenSearchSQLParser.g4

```diff
@@ -278,6 +278,7 @@ predicate
     : expressionAtom                                                #expressionAtomPredicate
     | left=predicate comparisonOperator right=predicate             #binaryComparisonPredicate
     | predicate IS nullNotnull                                      #isNullPredicate
+    | predicate BETWEEN predicate AND predicate                     #betweenPredicate
```

Comment:
I think we need factor our `highlight` and relevance functions so that queries like `SELECT * FROM index WHERE match_query(...) BETWEEN highlight(...) AND match_phrase(...)` are not allowed.

Also,  `BOOLEAN, BOOLEAN, BOOLEAN` version can be removed, I think. It only works if `from` and `to` are the same value. Ex: `SELECT * FROM phrases WHERE (4 in (1,4) BETWEEN TRUE AND TRUE`

Reply:
@MaxKsyunz Thanks for the preview comment!

1. As I understand, grammar file only defines the valid and recursive structure of a query. If we want to validate further, we may need to check, take `highlight()` for example, if it's in SELECT list in `Analyzer`.
2. Good point. I need to double check if any core type should be excluded.

Reply:
I refactored the implementation after reference other query engine code. Please check the PR description. Thanks!